metadata_layer.py

```diff
@@ -708,7 +708,7 @@ class MetaToModelUtility(object):
 
     log = logging.getLogger("Abstract metadata layer - mirror code")
 
-    def mirror_link(self, model_object, data_source, link, link_obj, policy):
+    def mirror_link(self, model_object, data_source, link, link_obj, policy, collection_id=None):
```

Comment:
Since the other arguments to this method are data model objects, `collection_id` should also be a data model object called `collection`.

Also, if `model_object` is a `LicensePool` then the only acceptable value for `collection` is `model_object.collection`. Anything else is wrong and should result in an exception or at least be ignored. That is, if a LicensePool is in collection A then it never makes sense to mirror its resources to collection B.

Reply:
Removed `collection_id`, added code for fetching collection dynamically:
```
# Mirror it.
collection = pools[0].collection if pools[0].collection else None
mirror.mirror_one(representation, mirror_to=mirror_url, collection=collection)
```